src/django/pfb_network_connectivity/settings.py

```diff
@@ -222,7 +222,7 @@
         'rest_framework.authentication.TokenAuthentication',
         'rest_framework.authentication.SessionAuthentication',
     ),
-    'DEFAULT_PAGINATION_CLASS': 'rest_framework.pagination.LimitOffsetPagination',
+    'DEFAULT_PAGINATION_CLASS': 'pfb_network_connectivity.pagination.OptionalLimitOffsetPagination',
```

Comment:
Yeah, hard to know whether we'll want no limits, restricted limits, or neither (we don't currently have a `max_limit` set on the default paginator) for other endpoints, but making the `limit=all` option opt-in makes sense.  I put back the default and set the modified one as the pagination class on NeighborhoodViewSet.